test/hclust.jl

```diff
@@ -141,4 +142,27 @@ end
     @test cutree(hA, h=0.9) == [1, 1]
 end
 
+@testset "Leaf ordering methods" begin
+    n = 10
+    mat = zeros(Int, n, n)
+
+    for i in 1:n
+        last = minimum([i+Int(floor(n/5)), n])
+        for j in i:last
+            mat[i,j] = 1
+        end
+    end
+
+    dm = pairwise(Euclidean(), mat, dims=2)
+
+    hcl1 = hclust(dm, linkage=:average)
+    hcl2 = hclust(dm, linkage=:average, leaforder=:barjoseph)
+
```

Comment:
Please also add the test for `:optimal` and for some incorrect value
```julia
@test_throws ArgumentError hclust(dm, linkage=:average, leaforder=:wrong)
```
And the tests with 0x0 and 1x1 distance matrices.